assets/js/components/view-switch-control/index.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { ButtonGroup, Button } from '@wordpress/components';
+import withComponentId from '@woocommerce/base-hocs/with-component-id';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+
+const ViewSwitchControl = ( {
+	className,
+	label = __( 'View', 'woo-gutenberg-products-block' ),
+	views = [],
+	selected = '',
+	onChange = () => {},
+	componentId,
+} ) => {
+	const classes = classnames( className, 'wc-block-view-switch-control' );
+	const htmlId = 'wc-block-view-switch-control-' + componentId;
+
+	return (
+		<div className={ classes }>
+			<label
+				htmlFor={ htmlId }
+				className="wc-block-view-switch-control__label"
+			>
+				{ label + ': ' }
+			</label>
+			<ButtonGroup id={ htmlId }>
+				{ views.map( ( view ) => (
+					<Button
+						key={ view.value }
+						isPrimary={ selected === view.value }
+						isLarge
+						aria-pressed={ selected === view.value }
+						onClick={ () => {
+							onChange( view.value );
+						} }
+					>
+						{ view.name }
+					</Button>
+				) ) }
+			</ButtonGroup>
+		</div>
+	);
+};
+
+ViewSwitchControl.propTypes = {
```

Comment:
What do you think about making `views` and `selected` required props? I can't think of a use case where they would be omitted.